src/System.Text.Json/src/System/Text/Json/Writer/UnicodeScalar.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers.Text
+{
+    /// <summary>
+    /// Represents a 24-bit Unicode scalar value.
+    /// A scalar value is any value in the range [U+0000..U+D7FF] or [U+E000..U+10FFFF].
+    /// </summary>
+    internal struct UnicodeScalar : IComparable<UnicodeScalar>, IEquatable<UnicodeScalar>
```

Comment:
Why can't this use the new `Rune` type?

Reply:
It can and should. This code (that I borrowed from corefxlab) wasn't updated to use `Rune`. I will fix this.